lightning/src/ln/data_persister.rs

```diff
@@ -0,0 +1,221 @@
+//! Logic for persisting data from ChannelMonitors on-disk. Per-platform data
+//! persisters are separated into the lightning-persist-data crate. These
+//! objects mainly interface with the ChainMonitor when a channel monitor is
+//! added or updated, and when they are all synced on startup.
+use chain::channelmonitor::{ChannelMonitor, ChannelMonitorUpdate, ChannelMonitorUpdateErr};
+use chain::keysinterface::ChannelKeys;
+use chain::transaction::OutPoint;
+use util::ser::{Writeable, Readable};
+
+use bitcoin::hash_types::{BlockHash, Txid};
+use bitcoin::hashes::hex::{ToHex, FromHex};
+use std::fs;
+use std::io::{Error, ErrorKind, Cursor};
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+/// ChannelDataPersister is responsible for persisting channel data: this could
+/// mean writing once to disk, and/or uploading to several backup services.
+///
+/// Note that for every new monitor, you **must** persist the new ChannelMonitor
+/// to disk/backups. And, on every update, you **must** persist either the
+/// ChannelMonitorUpdate or the updated monitor itself. Otherwise, there is risk
+/// of situations such as revoking a transaction, then crashing before this
+/// revocation can be persisted, then unintentionally broadcasting a revoked
+/// transaction and losing money. This is a risk because previous channel states
+/// are toxic, so it's important that whatever channel state is persisted is
+/// kept up-to-date.
+///
+/// Upon calls to update_channel_data, implementers of this trait should either
+/// persist the ChannelMonitorUpdate or the ChannelMonitor. If an implementer
+/// chooses to persist the updates only, they need to make sure that all the
+/// updates are applied to the ChannelMonitors *before* the set of channel
```

Comment:
nit: use double asterisks for markdown bold

Reply:
I was okay with just italics, but will update to be bold for uniformity.

Reply:
Oops! For some reason I imagined recalling seeing literal asterisks in our Rust docs and figured because two were needed. Feel free to leave as italics.